### ISSUE TITLE
Style/IfInsideElse and friends swallowed a condition when --safe-auto-correct'd

### ISSUE THREAD

**Issue by dhnaranjo** [rubocop-hq/rubocop#7235]
--safe-auto-correct of some less than ideal code resulted in a conditional branch completely getting deleted. Thankfully we had test covg around it, but I figured y'all would like to know. The code sample below is a direct copy of the code with object n method names changed.

--------

## Expected behavior

Rubocop either resolves issues without breaking functionality or identifies them without autocorrecting.

## Actual behavior

An entire conditional branch disappears

## Steps to reproduce the problem

Given code sample:
```
def rubocop_breaker
  if @thing && @thing.boolean_atty == false
    @thing = other_method(@thing.arg_one, @thing.arg_two)
  else
    if @thing && @thing.integer_atty && @thing.integer_atty < 100
      @thing = other_method(@thing.arg_one, @thing.arg_two)
    else
      @thing = @thing
    end
  end
end
```

Execute Rubocop autocorrect and receive:

```
def rubocop_breaker
  @thing = if @thing && @thing.boolean_atty == false
    other_method(@thing.arg_one, @thing.arg_two)
  else
    @thing = @thing
  end
end
```

As you see, the `@thing.integer_atty < 100` branch just... disappeared.

With a fix output of:
```
app/rubocop_breaker.rb:25:3: C: Style/ConditionalAssignment: Use the return of the conditional for variable assignment and comparison.
  if @thing && @thing.boolean_atty == false ...
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
app/rubocop_breaker.rb:28:5: C: Style/ConditionalAssignment: Use the return of the conditional for variable assignment and comparison.
    if @thing && @thing.integer_atty && @thing.integer_atty < 100 ...
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
app/rubocop_breaker.rb:28:5: C: Style/IfInsideElse: Convert if nested inside else to elsif.
    if @thing && @thing.integer_atty && @thing.integer_atty < 100
    ^^
app/rubocop_breaker.rb:28:8: C: Style/SafeNavigation: Use safe navigation (&.) instead of checking if an object exists before calling the method.
    if @thing && @thing.integer_atty && @thing.integer_atty < 100

```

## RuboCop version

```
$ [bundle exec] rubocop -V
0.72.0 (using Parser 2.6.3.0, running on ruby 2.4.5 x86_64-darwin18)
```

**Comment by buehmann** [comment]
Here's a reduced example:
```ruby
if first
  var = 1
else
  if second
    var = 2
  else
    var = 3
  end
end

if first
  var = 1
elsif second
  var = 2
else
  var = 3
end
```
is autocorrected by `Style/ConditionalAssignment` to
```ruby
var = if first
  1
else
  var = 3
      end

var = if first
  1
elsif second
  2
else
  3
      end
```
It does not fail on the `elsif` variant. The cop might be fooled by the similar ASTs.

**Comment by dhnaranjo** [comment]
Dang y'all, good looking out with the quick response. Appreciate all your efforts.
